tango/gevent_executor.py

```diff
@@ -89,14 +85,14 @@ def get_with_exception(result, block=True, timeout=None):
     return result
 
 
-def spawn(threadpool, fn, *args, **kwargs):
-    # The gevent threadpool do not raise exception with async results,
-    # we have to wrap it
-    fn = wrap_errors(fn)
-    result = threadpool._spawn(fn, *args, **kwargs)
-    result._get = result.get
-    result.get = types.MethodType(get_with_exception, result, type(result))
-    return result
+class ThreadPool(gevent.threadpool.ThreadPool):
+
+    def spawn(self, fn, *args, **kwargs):
+        fn = wrap_errors(fn)
+        result = super(ThreadPool, self).spawn(fn, *args, **kwargs)
+        result._get = result.get
+        result.get = types.MethodType(get_with_exception, result, type(result))
```

Comment:
`type(result)` argument needs to be removed, it's not compatible with python 3 (and it's not mandatory with python 2)